calculator/main.py

```diff
@@ -56,20 +90,27 @@ def get_history(calc):
         (:execution_time, :operation_name, :params, :result)
 
         ie:
-        ('2016-05-20 12:00:00', 'add', (1, 2), 3),
+        ('2016-05-20 12:00:00', 'add', (1, 2), 3)
     """
-    pass
+    return calc['history']
 
 
 def reset_history(calc):
     """
     Resets the calculator history back to an empty list.
     """
-    pass
+    calc['history'] = []
 
 
 def repeat_last_operation(calc):
     """
     Returns the result of the last operation executed in the history.
     """
-    pass
+    
+    # If History key in calc Dictionary has content, rerun last operation.
+    if calc['history']:
+        operation = calc['history'][-1][1]
+        params = calc['history'][-1][2]
+        return calc["operations"][operation](*params)
```

Comment:
As we are storing the result of the operation in the history, it's not necessary to re run the operation. We can just return the result.